<?xml version="1.0"?>
<launch>
  <!-- MoveIt options -->
  <arg name="pipeline" default="ompl" doc="Planning pipeline to use with MoveIt"/>

  <!-- Gazebo options -->
  <arg name="gazebo_gui" default="true" doc="Start Gazebo GUI"/>
  <arg name="paused" default="false" doc="Start Gazebo paused"/>
  <arg name="world_name" default="worlds/empty.world" doc="Gazebo world file"/>
  <arg name="world_pose" default="-x 0 -y 0 -z 0 -R 0 -P 0 -Y 0" doc="Pose to spawn the robot at"/>
  <param name="use_sim_time" value="true"/>
  <!-- <param name="april_box_description" command="$(find xacro)/xacro '$(find rh_moveit)/apriltags/Apriltag36_11_00001/model.sdf'" /> -->


  <!-- Launch Gazebo and spawn the robot -->
  <include file="$(dirname)/gazebo.launch" pass_all_args="true"/>

  <!-- Launch MoveIt -->
  <include file="$(dirname)/demo.launch" pass_all_args="true">
    <!-- robot_description is loaded by gazebo.launch, to enable Gazebo features -->
    <arg name="load_robot_description" value="false" />
    <arg name="moveit_controller_manager" value="ros_control" />
  </include>

  <include file="$(dirname)/ur5.launch" pass_all_args="true"/>

  <!-- Add the box with AprilTag -->
  <!-- <node name="spawn_april_box" pkg="gazebo_ros" type="spawn_model" args="-urdf -param april_box_description -model Apriltag36_11_00001" /> -->

  <node pkg="apriltag_ros" type="apriltag_ros_continuous_detector" name="tag_detector" output="screen">
    <param name="tag_family" value="tag36h11"/>
    <param name="camera_name" value="camera"/>
    <param name="camera_frame" value="left_camera_frame"/>
    <param name="image_topic" value="/left_arm_camera/left_camera/image_raw"/>
    <param name="camera_info_topic" value="/left_arm_camera/left_camera/camera_info"/>
    <param name="tag_size" value="0.2"/>
  </node>

  

  <!-- Spawn the cameras -->
  <!-- <param name="cameras_description" command="$(find xacro)/xacro '$(find rh_moveit)/config/cameras.urdf'" />
  <node name="spawn_cameras" pkg="gazebo_ros" type="spawn_model" args="-param cameras_description -urdf -model cameras" /> -->
</launch>
